app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -347,8 +382,16 @@ watchEffect(() => {
     columnDefs = [toField('Value')]
     rowData = [{ Value: toRender(data_.json) }]
   } else {
-    const indicesHeader = ('indices_header' in data_ ? data_.indices_header : []).map(toField)
-    const dataHeader = ('header' in data_ ? data_.header : [])?.map(toField) ?? []
+    const indicesHeader = ('indices_header' in data_ ? data_.indices_header : []).map((v) =>
+      toField(v),
+    )
+    const dataHeader =
+      ('header' in data_ ? data_.header : [])?.map((v, i) => {
+        const valueType =
+          data_.value_type && data_.value_type[i] ? (data_.value_type[i]?.constructor as any) : null
```

Comment:
`constructor` is not a string, but a reference to the construcing function (e.g. class). You should explicitly print it here. Also, the `?.` operator works for index access, the syntax is just a little wierd.
```suggestion
        const constructor = data_.value_type?.[i]?.constructor
        const valueType = constructor  ? `${constructor}` : null
```

Reply:
`constructor` is a string field in our serialized form of the `Value_Type`, e.g. `data_.value_type[i]` will be for example `{"type": "Value_Type", "constructor": "Integer", "bits": 64}`.

Reply:
So the `${constructor}` probably shouldn't be needed I think?

Reply:
Yeah, I was a little confused. Already edited the top post to reflect that. Still a better guard would be good here, casting to `any` here is totally unnecessary.